crates/ide_assists/src/handlers/convert_iter_for_each_to_for.rs

```diff
@@ -0,0 +1,229 @@
+use ide_db::helpers::FamousDefs;
+use syntax::{
+    ast::{self, edit::AstNodeEdit, make, ArgListOwner},
+    AstNode,
+};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+/// Assist: convert_iter_for_each_to_for
+//
+/// Converts an Iterator::for_each function into a for loop.
+///
+/// ```rust
+/// fn main() {
+///     let vec = vec![(1, 2), (2, 3), (3, 4)];
+///     x.iter().for_each(|(x, y)| {
+///         println!("x: {}, y: {}", x, y);
+///    })
+/// }
+/// ```
+/// ->
+/// ```rust
+/// fn main() {
+///     let vec = vec![(1, 2), (2, 3), (3, 4)];
+///     for (x, y) in x.iter() {
+///         println!("x: {}, y: {}", x, y);
+///     });
+/// }
+/// ```
+pub(crate) fn convert_iter_for_each_to_for(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let method;
+
+    let stmt = if let Some(stmt) = ctx.find_node_at_offset::<ast::ExprStmt>() {
+        method = ast::MethodCallExpr::cast(stmt.syntax().first_child()?)?;
+        Some(stmt)
+    } else {
+        method = match ctx.find_node_at_offset::<ast::Expr>()? {
+            ast::Expr::MethodCallExpr(expr) => expr,
+            ast::Expr::ClosureExpr(expr) => {
+                ast::MethodCallExpr::cast(expr.syntax().ancestors().nth(2)?)?
+            }
+            _ => {
+                return None;
+            }
+        };
+        None
+    };
```

Comment:
Correct me if I'm wrong but I feel like this should work

```suggestion
	let method = ctx.find_node_at_offset::<ast::MethodCallExpr>()?;
	let stmt = method.syntax().parent().and_then(ast::ExprStmt::cast);
```